multimap/src/main/java/org/infinispan/tutorial/simple/multimap/InfinispanMultimap.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.tutorial.simple.multimap;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.multimap.api.embedded.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.embedded.MultimapCache;
+import org.infinispan.multimap.api.embedded.MultimapCacheManager;
+
+public class InfinispanMultimap {
+
+   public static void main(String[] args) {
+      // Construct a local cache manager with default configuration
+      DefaultCacheManager cacheManager = new DefaultCacheManager();
+
+      // Obtain a multimap cache manager from the regular cache manager
+      MultimapCacheManager multimapCacheManager = EmbeddedMultimapCacheManagerFactory.from(cacheManager);
+
+      // Define de multimap cache configuration, as a regular cache
+      multimapCacheManager.defineConfiguration("multimap", new ConfigurationBuilder().build());
+
+      // Get the MultimapCache
+      MultimapCache<String, String> multimap = multimapCacheManager.get("multimap");
+
+      // Store multiple values in a key
+      multimap.put("key", "value1");
+      multimap.put("key", "value2");
+      multimap.put("key", "value3");
+
+      // Retrieve the values
+      multimap.get("key").whenComplete((values, ex) -> {
```

Comment:
There's no guarantee `get` will be called before the puts have complete. This could confuse users. 

You should take the put returns, create a a CompletableFuture of them by calling `CompletableFuture.allOf`, and when that completes do the get. That's the only way to make this tutorial reliable.

Reply:
Agreed. Something like the following should work. Or you could even chain the get in `whenComplete` of the puts.

`CompletableFuture.allOf(multimap.put("key", "value1"), multimap.put("key", "value2") ... ).join()`

Reply:
Thank you, it was super fast in my laptop and did not realize about it !